src/main/java/com/swervedrivespecialties/exampleswerve/commands/DriveCommand.java

```diff
@@ -2,22 +2,28 @@
 
 import com.swervedrivespecialties.exampleswerve.Robot;
 import com.swervedrivespecialties.exampleswerve.subsystems.DrivetrainSubsystem;
-import edu.wpi.first.wpilibj.command.Command;
+import com.swervedrivespecialties.exampleswerve.util.util;
+
 import edu.wpi.first.wpilibj.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
 import org.frcteam2910.common.robot.Utilities;
 
-public class DriveCommand extends Command {
+public class DriveCommand extends CommandBase {
+
+    private static DrivetrainSubsystem _drive;
 
-    public DriveCommand() {
-        requires(DrivetrainSubsystem.getInstance());
+    public DriveCommand(DrivetrainSubsystem drive) {
```

Comment:
This is a fine approach, but if I were you I'd ditch this. I think going back to doing everything with Singletons will clean up the code quite a bit

Reply:
Ok, I'll change that